src/accessor.ts

```diff
@@ -137,15 +132,19 @@ export class IACAccessor {
    * @param scanTimeOut max time period for which scanning should be attempted.
    * @param scanStartTime timestamp in millis at which scanning was stared.
    * @param version action version, use to contruct user agent.
+   * @param httpClient http client to interact with IAC Scanning API.
+   * @param date used for retrieval of current time.
    */
   constructor(
     private readonly baseURL: string,
     private readonly organizationId: string,
     private readonly scanTimeOut: number,
     private readonly scanStartTime: number,
     private readonly version: string,
+    private readonly httpClient?: HttpClient,
+    private readonly date?: Date,
   ) {
-    this.client = new HttpClient(USER_AGENT(version));
+    this.client = httpClient ?? new HttpClient(USER_AGENT(version));
```

Comment:
Why this change?

Reply:
We want to mock the http client in our UT's. In order to do that, i have made http client an optional param to the constructor, so that the mocked http client could be passed from the UT's.

Reply:
You can pretty easily mock `HttpClient.prototype` without needing the dependency injection.

Reply:
I believe this is inline with testing best practices that we inject any dependency so that it's unit testable.

Reply:
Removed the dependency, we have also removed sinon from UT's now.